src/create-slice.ts

```diff
@@ -1,5 +1,5 @@
 type SliceActions<Value> = {
-  [actionName: string]: (...args: never[]) => (prev: Value) => Value;
+  [actionName: string]: (...args: never[]) => (prev: Value) => Value | void;
```

Comment:
I don't think returning void should be allowed for the normal createSlice.
```suggestion
  [actionName: string]: (...args: never[]) => (prev: Value) => Value;
```

Reply:
Agreed, allowing void as a return type in `createSlice` was an oversight in my attempt to unify the index signatures for compatibility with `withSlices`. The original intent was to ensure type safety for both `createSlice` and `createSliceWithImmer`, as they have differing return types which `withSlices` must accommodate. To resolve this, I've introduced distinct index signatures for each slice creation method, ensuring that `withSlices` correctly handles the types without incorrectly allowing void returns in `createSlice`.